src/streamlink/plugins/aloula.py

```diff
@@ -0,0 +1,113 @@
+"""
+$description Live TV channels and video on-demand service from the SBA, a Saudi, state-owned broadcaster.
+$url aloula.sa
+$type live, vod
+"""
+
+import logging
+import re
+
+from streamlink.plugin import Plugin, pluginmatcher
+from streamlink.plugin.api import validate
+from streamlink.stream.hls import HLSStream
+
+log = logging.getLogger(__name__)
+
+
+@pluginmatcher(re.compile(r"""
+    https?://(?:www\.)?aloula\.sa/(?:\w{2}/)?
+    (?:
+        live/(?P<live_slug>[^/?&]+)
+    |
+        episode/(?P<vod_id>\d+)
+    )
+""", re.VERBOSE))
+class Aloula(Plugin):
+    def get_live(self, live_slug):
+        live_data = self.session.http.get(
+            "https://aloula.faulio.com/api/v1/channels",
+            schema=validate.Schema(
+                validate.parse_json(),
+                [{
+                    "id": int,
+                    "url": str,
+                    "title": str,
+                    "has_live": bool,
+                    "has_vod": bool,
+                    "streams": {
+                        "hls": validate.url(),
+                    },
+                }],
+                validate.filter(lambda k: k["url"] == live_slug),
+            ),
+        )
+        if not live_data:
+            return
+        live_data = live_data[0]
+        log.trace(f"{live_data!r}")
+
+        if not live_data["has_live"]:
+            log.error("Stream is not live")
+            return
+
+        self.id = live_data["id"]
+        self.author = "SBA"
+        self.title = live_data["title"]
+        self.category = "Live"
+        return HLSStream.parse_variant_playlist(self.session, live_data["streams"]["hls"])
+
+    def get_vod(self, vod_id):
+        vod_data = self.session.http.get(
+            f"https://aloula.faulio.com/api/v1/video/{vod_id}",
+            acceptable_status=(200, 401),
+            schema=validate.Schema(
+                validate.parse_json(),
+                validate.any(
+                    validate.all(
+                        {"blocks": [{
+                            "id": str,
+                            "program_title": str,
+                            "title": str,
+                            "season_number": int,
+                            "episode": int,
+                        }]},
+                        validate.get(("blocks", 0)),
+                    ),
+                    {"cms_error": str, "message": str},
+                ),
+            ),
+        )
+
+        log.trace(f"{vod_data!r}")
+        if "cms_error" in vod_data and vod_data["cms_error"] == "auth":
+            log.error("This stream requires a logged-in session cookie to be supplied")
```

Comment:
This could be more specific. The headers which are required are `authorization` and `profile`. Adding plugin parameters for this is probably too much though for just vods.

Reply:
I could just not be lazy and implement the ability to login.  The reason I didn't was partly through laziness and partly the belief that it probably won't get used.  The text above was just based on a guess.  It's either implement the login properly or perhaps just change this to `This stream requires a login; specify appropriate authorization and profile HTTP headers`.